devito/nodes.py

```diff
@@ -473,9 +474,19 @@ def __init__(self, name, body, retval, parameters=None, prefix=('static', 'inlin
         self.name = name
         self.body = as_tuple(body)
         self.retval = retval
-        self.parameters = as_tuple(parameters)
         self.prefix = prefix
 
+        if all([isinstance(x, RuntimeArgProvider) for x in parameters]):
+            args = flatten([x.rtargs for x in parameters])
+        else:
+            assert(all([isinstance(x, RuntimeArgument) for x in parameters]))
+            args = parameters
+
+        self.parameters = as_tuple(args)
+
+        self.t_args = [x for x in args if isinstance(x, TensorArgument)]
```

Comment:
Following Devito's and Sympy's conventions, can we add ``is_TensorArgument``-like fields to Argument types ? So that we don't need to use ``isinstance`` **and** we implicitly convey the message that what we have at this point is an object of **that** hierarchy